### PR TITLE
Add support for mirrors and sources in Key Value Store

### PR DESCRIPTION
**Pull request by Jarema**

### CODE REVIEW

File: async-nats/src/jetstream/context.rs [hunk 1]
```diff
@@ -192,13 +192,43 @@ impl Context {
...
     {
-        let config: Config = stream_config.into();
+        let mut config: Config = stream_config.into();
         if config.name.is_empty() {
             return Err(Box::new(io::Error::new(
                 ErrorKind::InvalidInput,
                 "the stream name must not be empty",
             )));
         }
+        if let Some(ref mut mirror) = config.mirror {
```
[the remark this answers is not in the document]
Comment (Jarema) [reply async-nats/src/jetstream/context.rs:202]: I don't see how flattening this with combinators makes it easier to read. Care to show an example?

File: async-nats/src/jetstream/context.rs [hunk 2]
```diff
@@ -192,13 +192,43 @@ impl Context {
...
+                        "domain and external are both set",
+                    )));
+                }
+                mirror.external = Some(External {
+                    api_prefix: format!("$JS.{}.API", domain),
+                    delivery_prefix: None,
+                })
+            }
+        }
+        if let Some(ref mut sources) = config.sources {
```
[the remark this answers is not in the document]
Comment (Jarema) [reply async-nats/src/jetstream/context.rs:216]: I don't see how flattening this with combinators makes it easier to read. Care to show an example?

[the remark this answers is not in the document]
Comment (caspervonb) [reply async-nats/src/jetstream/context.rs:245]: Can revisit and clean up later.

**Review by caspervonb — APPROVED** [review]
lgtm

**Comment by Jarema** [comment]
addresses https://github.com/nats-io/nats-architecture-and-design/issues/167

**Pull request merged**